### PR TITLE
Fix counts and size sum in hotbackup META files APM-409

### PR DESCRIPTION
**Pull request by neunhoef**
This is a small bug fix to get the counts and file size sum in hotbackup
META files correct. The problem was that due to a bug in the accounting
directories have been counted, rendering the sums incorrect. This PR
is accompanied by an enterprise PR and together they fix the sums and
add a new filed to recognize new style META files.

- Get counts in hotbackup META files right.
- CHANGELOG.

### Scope & Purpose

*(Please describe the changes in this PR for reviewers, motivation, rationale - **mandatory**)*

- [*] :hankey: Bugfix

### Checklist

- [x] Tests
  - [x] **integration tests**, see enterprise repo
  - [x] **release test automation** also has a test now
- [x] :book: CHANGELOG entry made
- [x] Backports
  - [x] Backport for 3.9: *(Please link PR)*
  - [x] Backport for 3.8: *(Please link PR)*
  - [x] Backport for 3.7: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [x] Enterprise PR: https://github.com/arangodb/enterprise/pull/886
- [x] GitHub issue / Jira ticket: https://arangodb.atlassian.net/browse/APM-409
      This is the first quick step towards fulfilling this ticket.

### CODE REVIEW

**Review by maierlars — APPROVED** [review]

File: tests/js/client/shell/shell-hotbackup-noncluster.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
+/*jshint globalstrict:false, strict:false */
```
Comment (dothebart) [inline tests/js/client/shell/shell-hotbackup-noncluster.js:1]: shouldn't this test be in the enterprise repository?

File: tests/js/client/shell/shell-hotbackup-noncluster.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    },
+
+////////////////////////////////////////////////////////////////////////////////
+/// Check sums and counts in META after local upload
+////////////////////////////////////////////////////////////////////////////////
+
+    testCheckSumsAndCounts: function() {
+      let fs = require("fs");
+      let tempPath = fs.getTempPath();
+      console.warn("Temporary path:", tempPath);
```
Comment (dothebart) [inline tests/js/client/shell/shell-hotbackup-noncluster.js:104]: please don't use WARN for non-emergency logging.

File: tests/js/client/shell/shell-hotbackup-noncluster.js (hunk 1 above)
Comment (neunhoef) [reply tests/js/client/shell/shell-hotbackup-noncluster.js:1]: Yes, it should, will move, once it works in Jenkins.

**Review by dothebart — APPROVED** [review]

File: arangod/StorageEngine/HotBackupCommon.h [hunk 3]
```diff
@@ -95,6 +98,7 @@ struct BackupMeta {
       }
       builder.add(POTENTIALLYINCONSISTENT,
                   VPackValue(_potentiallyInconsistent));
+      builder.add(COUNTINCLUDESFILESONLY, VPackValue(true));
```
Comment (jsteemann) [inline arangod/StorageEngine/HotBackupCommon.h:101]: Shouldn't this be
```suggestion
      builder.add(COUNTINCLUDESFILESONLY, VPackValue(_countIncludeFilesOnly));
```

**Pull request merged**